src/main/java/racingcar/controller/GameController.java

```diff
@@ -0,0 +1,65 @@
+package racingcar.controller;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import racingcar.movableStrategy.MoveStrategy;
+import racingcar.modle.vehicle.Vehicles;
+import racingcar.view.InputView;
+import racingcar.view.OutputView;
+
+public class GameController {
+    private final Vehicles vehicles;
+    private final List<Vehicles> eachRoundResults;
+    private int playCount;
+
+    public GameController(MoveStrategy moveStrategy) {
+        this.vehicles = new Vehicles(moveStrategy);
+        this.eachRoundResults = new ArrayList<>();
+    }
+
+    public void gameSetting() {
+        OutputView.requestInputPlayerNames();
+        vehicles.nameSettings(InputView.playerNames());
+        OutputView.requestPlayCount();
+        playCount = InputView.playCount();
+    }
+
+    public void gamePlay() {
+        OutputView.printResultMessage();
+        playEachRound(playCount);
+    }
+
+    private void playEachRound(int playCount) { //흠...이거너무 찝찝한데
+        eachRoundResults.add(vehicles);
+        for (int i = 1; i < playCount + 1; i++) { //흠...이부분을 어떻게 더 깔끔하게 리팩터링할 수 있을까?...
+            Vehicles thisRoundVehicles = eachRoundResults.get(i - 1).moveAll();
+            eachRoundResults.add(thisRoundVehicles);
+            printRoundResult(thisRoundVehicles);
+        }
+    }
+
+    private void printRoundResult(Vehicles RoundVehicles) {
+        OutputView.printRoundMovementGraph(RoundVehicles.names(), RoundVehicles.positions());
+    }
+
+    public void gameResult() {
+        OutputView.printWinner(findWinner(eachRoundResults.getLast()));
```

Comment:
아래와 같이 표현하면 어떨까요?? 가독성 및 유지보수에도 용이하고 디버깅을 할 때도 훨씬 좋을 것 같다는 생각이 드는데, 형석님 의견은 어떠신가요?

```suggestion
        Vehicles vehicles = eachRoundResults.getLast();
        OutputView.printWinner(findWinner(vehicles)));
```

Reply:
> 아래와 같이 표현하면 어떨까요?? 가독성 및 유지보수에도 용이하고 디버깅을 할 때도 훨씬 좋을 것 같다는 생각이 드는데, 형석님 의견은 어떠신가요?

이야~ 너무 좋은거 같습니다.
코드를 보고 의미가 바로 전달되네요!!

가로안에 계속 가로를 넣어서 파고드는게 찝찝했는데!! 의견 감사합니다!😊